config/jobs/ppc64le-cloud/build-docker/postsubmit-build-docker.yaml

```diff
@@ -0,0 +1,149 @@
+postsubmits:
+ ppc64le-cloud/docker-ce-build:
+    - name: postsubmit-build-docker
+      cluster: k8s-ppc64le-cluster
+      decorate: true
+      decoration_config:
+        timeout: 3h
+      branches:
+          - main
+      run_if_changed: 'env/env.list'
+      reporter_config:
+        slack:
+          channel: 'prow-job-notifications'
+          report_template: 'Job {{.Spec.Job}} of type {{.Spec.Type}} ended with state {{.Status.State}}. <!subteam^S02N6DWBX0F> <{{.Status.URL}}|View logs>'
+      spec:
+        nodeSelector:
+          dedicated: runtimesTeam
+        tolerations:
+        - key: dedicated
+          operator: "Equal"
+          value: runtimesTeam
+          effect: "NoSchedule"
+        containers:
+        - image: quay.io/powercloud/docker-ce-build
+          command:
+          - /bin/bash
+          args:
+          - -c
+          - |
+            set -o errexit
+            set -o nounset
+            set -o pipefail
+            set -o xtrace
+
+            # Set the SSH key for git from the secret 'git-ssh-docker-ce-build'
+            echo "* Set up the ssh key for git *"
+            mkdir -p ~/.ssh
+            cp /etc/ssh-key/ssh-privatekey  ~/.ssh/id_rsa
+            chmod 600 ~/.ssh/id_rsa
+            ssh-keyscan -t rsa github.com >> ~/.ssh/known_hosts
+
+            echo "* Start prow-build-docker *"
+            ./prow-build-docker.sh
+            rc=$?
+            [ $rc != 0 ] && echo "ERROR: prow-build-docker exited with code:$rc"
+            exit $rc
+
+          securityContext:
+            privileged: true
+          volumeMounts:
+          - name: docker-root
+            mountPath: /var/lib/docker
+          - name: ssh-key
+            readOnly: true
+            mountPath: /etc/ssh-key
+          env:
+            - name: S3_SECRET_AUTH
+              valueFrom:
+                secretKeyRef:
+                  name: docker-s3-credentials
+                  key: password
+        volumes:
+        - name: docker-root
+          emptyDir: {}
+        - name: ssh-key
+          secret:
+           secretName: git-ssh-docker-ce-build
+
+    - name: postsubmit-build-test-containerd
+      cluster: k8s-ppc64le-cluster
+      decorate: true
+      decoration_config:
+        timeout: 3h
+      run_if_changed: 'job/postsubmit-build-docker'
+      branches:
+        - prow-job-tracking
```

Comment:
@alunsin Can you please explain the need to have a separate branch for prow job tracking?

IMO, we could reduce the length of these prow job yamls and make it simple if the `TIMESTAMP_FILE` is in the main branch itself.

Reply:
@alunsin I see the `date.list` file in the master branch at https://github.com/ppc64le-cloud/docker-ce-build/blob/main/env/date.list
created for a similar purpose as the `postsubmit-build-docker` file in the `prow-job-tracking` branch.
Would it not be needed to update `date.list` file in the `master` branch? (I see it having a old value)

In fact, the script at https://github.com/ppc64le-cloud/docker-ce-build/blob/main/prow-build-docker.sh#L58 is using the `date.list` file in the `main` branch as a temporary location before copying to the `prow-job-tracking` branch.

Also, the folder `env` in the `main` branch has only the job triggering files that we update to spin another prow job.

I see isolation already present in terms of having a separate folder. Would it be right to rename `env` folder in the main branch to `job`?

Reply:
The `date.list` is a bit messy in term of design and this something that we could review and improve in the future.
I need to double check, but I am  pretty sure the `env/date.list` file can be removed from git.

If you look at [https://github.com/ppc64le-cloud/docker-ce-build/blob/87fc960fa9acffb91e8e55f9c7529582618479e5/prow-build-docker.sh#L10](/prow-build-docker.sh) , the date.list file is updating locally  in the `postsubmit-build-docker` job, which is the 1st job that will be spawned.
Then we transfer this date content from date.list file to the secondaries jobs by copying this content into the file postsubmit-build-docker in the `prow-job-tracking` branch. The other jobs uses that file to populate the date.list file and make sure they are all using the same s3 directory.
To sum up I would prefer removing the `env/date.list` from the git repo. Will test that as soon as I am done with the current docker refresh.